### ISSUE TITLE
No directionality Widget found

### ISSUE THREAD

**Issue by clickglue** [gskinnerTeam/flutter-stats-fl#3]

**Comment by clickglue** [comment]
Get this error, caused by StatsFl: "No directionality Widget found"

Wrapped MyApp() in StatsFl like so:

void main() => runApp(
      StatsFl(
        isEnabled: true, //Toggle on/off
        width: 400, //Set size
        height: 32, //
        maxFps: 90, // Support custom FPS target (default is 60)
        showText: true, // Hide text label
        sampleTime: .5, //Interval between fps calculations, in seconds.
        totalTime: 15, //Total length of timeline, in seconds.
        //align: Alignment.topLeft,
        //textDirection: TextDirection.ltr, //Alignment of statsbox
        child: MyApp(),
      ),
    );

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => _MyAppState();
}

**Comment by jing-pei** [comment]
same problem

<details>
  <summary> flutter doctor -v </summary>

```
[✓] Flutter (Channel beta, 1.21.0-9.2.pre, on macOS 11.0 20A5364e, locale
    zh-Hans-CN)
    • Flutter version 1.21.0-9.2.pre at /Users/angus/flutter
    • Framework revision 81a45ec2e5 (10 days ago), 2020-08-27 14:14:33 -0700
    • Engine revision 20a9531835
    • Dart version 2.10.0 (build 2.10.0-7.3.beta)
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/angus/Library/Android/sdk
    • Platform android-R, build-tools 29.0.2
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.6, Build version 11E708
    • CocoaPods version 1.9.0.beta.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 49.0.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] IntelliJ IDEA Ultimate Edition (version 2019.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 43.0.3
    • Dart plugin version 193.6015.53

[✓] VS Code (version 1.48.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.14.0

[✓] Connected device (3 available)
    • iPhone 11 Pro (mobile) • 468F375E-3F71-48C9-A6B3-DC0B41368118 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-13-6 (simulator)
    • Web Server (web)       • web-server                           •
      web-javascript • Flutter Tools
    • Chrome (web)           • chrome                               •
      web-javascript • Google Chrome 85.0.4183.83

• No issues found!

```

</details>

```
Waiting for iPhone 11 Pro to report its views...
Debug service listening on ws://127.0.0.1:58775/9xy7UkdYzmA=/ws
Syncing files to device iPhone 11 Pro...

════════ Exception caught by widgets library ═══════════════════════════════════════════════════════
The following assertion was thrown building DefaultTextStyle(debugLabel: (englishLike body1 2014).merge(blackCupertino bodyText2), inherit: false, color: Color(0xdd000000), family: .SF UI Text, size: 14.0, weight: 400, baseline: alphabetic, decoration: TextDecoration.none, softWrap: wrapping at box width, overflow: clip):
No Directionality widget found.

Stack widgets require a Directionality widget ancestor to resolve the 'alignment' argument.


The default value for 'alignment' is AlignmentDirectional.topStart, which requires a text direction.

The specific widget that could not find a Directionality ancestor was: Stack
  alignment: AlignmentDirectional.topStart
  fit: loose
  dirty
The ownership chain for the affected widget is: "Stack ← DefaultTextStyle ← AnimatedDefaultTextStyle ← _InkFeatures-[GlobalKey#265df ink renderer] ← NotificationListener<LayoutChangedNotification> ← PhysicalModel ← AnimatedPhysicalModel ← Material ← StatsFl ← [root]"

Typically, the Directionality widget is introduced by the MaterialApp or WidgetsApp widget at the top of your application widget tree. It determines the ambient reading direction and is used, for example, to determine how to lay out text, how to interpret "start" and "end" values, and to resolve EdgeInsetsDirectional, AlignmentDirectional, and other *Directional objects.
Instead of providing a Directionality widget, another solution would be passing a non-directional 'alignment', or an explicit 'textDirection', to the Stack.

The relevant error-causing widget was: 
  Material file:///Users/angus/.pub-cache/hosted/pub.flutter-io.cn/statsfl-1.1.0+2/lib/statsfl.dart:116:12
When the exception was thrown, this was the stack: 
#0      debugCheckHasDirectionality.<anonymous closure> (package:flutter/src/widgets/debug.dart:267:7)
#1      debugCheckHasDirectionality (package:flutter/src/widgets/debug.dart:287:4)
#2      Stack._debugCheckHasDirectionality (package:flutter/src/widgets/basic.dart:3340:14)
#3      Stack.createRenderObject (package:flutter/src/widgets/basic.dart:3352:12)
#4      RenderObjectElement.mount (package:flutter/src/widgets/framework.dart:5428:28)
...
```

**Comment by esDotDev** [comment]
Should be fixed in 1.1.0+3

**Comment by talhachattha162** [comment]
use parenthesis after arrow operator and  closing parenthesis at end of function because you are using multiple statements
